Tests/MongoSwiftSyncTests/MongoCollectionTests.swift

```diff
@@ -589,6 +589,66 @@ final class MongoCollectionTests: MongoSwiftTestCase {
             .to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
 
+    func testFindAndModifyWithHint() throws {
+        guard let client = _client else {
```

Comment:
This file is kind of old, so it uses some of our older testing mechanisms (e.g. shared client / collection on `self`). These days we've moved away from this and have instead opted to create new clients / dbs / collections per test to avoid sharing state and to keep things encapsulated. To that end, could we update this test to make use of `withTestNamespace` instead of `_client` and `self.coll`?

Reply:
done